### PR TITLE
feat: add custom omit utility type

### PR DESCRIPTION
**Pull request by cf-remylenoir**
# Purpose of PR

We need this custom Omit utility type to support upcoming library updates where some extended-type properties get lost when using Omit.

## Reading 🍿 
- https://github.com/microsoft/TypeScript/issues/36981
- https://github.com/microsoft/TypeScript/issues/54451

### CODE REVIEW

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -71,3 +71,15 @@ export type PickUnion<UnionType, Keys> = Exclude<
   UnionType,
   Exclude<UnionType, Keys>
 >;
+
+/**
+ * Also known as mapped type or homomorphic.
```
Comment (cf-remylenoir) [inline packages/core/src/types.ts:76]:
```suggestion
 * Also known as mapped omit or homomorphic.
```

**Review by denkristoffer — COMMENTED** [review]
You don't want to keep the name `MappedOmit`?

**Comment by cf-remylenoir** [comment]
> You don't want to keep the name `MappedOmit`?

I don't have a strong opinion on this one, I renamed it.

**Review by denkristoffer — APPROVED** [review]

**Comment by cf-remylenoir** [comment]
Ugh, I forgot to export the actual type from the barrel file 😄

**Pull request merged**